public/app/plugins/datasource/postgres/sqlUtil.ts

```diff
@@ -0,0 +1,146 @@
+import { isEmpty } from 'lodash';
+
+import { RAQBFieldTypes, SQLExpression, SQLQuery } from 'app/features/plugins/sql/types';
+
+export function getRAQBType(type: string): RAQBFieldTypes {
+  switch (type) {
+    case 'boolean': {
+      return 'boolean';
+    }
+    case 'bit':
+    case 'bit varying':
+    case 'character':
+    case 'character varying':
+    case 'text': {
+      return 'text';
+    }
+    case 'smallint':
+    case 'integer':
+    case 'bigint':
+    case 'decimal':
+    case 'numeric':
+    case 'real':
+    case 'double precision':
+    case 'serial':
+    case 'bigserial':
+    case 'smallserial': {
+      return 'number';
+    }
+    case 'date': {
+      return 'date';
+    }
+    case 'time':
+    case 'time with time zone':
+    case 'time without time zone':
+    case 'interval': {
+      return 'time';
+    }
+    case 'timestamp':
+    case 'timestamp with time zone':
+    case 'timestamp without time zone': {
+      return 'datetime';
+    }
+    default:
+      return 'text';
+  }
+}
+
+export function getIcon(type: string) {
+  switch (type) {
+    case 'boolean': {
```

Comment:
this function looks redundant.  maybe we should return {type: string, icon: string} from getRAQBType.  then we don't need this getIcon function

Reply:
That makes sense!